background.js

```diff
@@ -0,0 +1,108 @@
+var urls = {};
+var currentUrl = null;
+var currentDay = null;
+var startTime = null;
+const MAX_TOP_URLS = 7;
+
+function getTopUrls() {
+    updateCurrentTime();
+	var items = sortUrls(urls);
+    return items.slice(0, Math.Min(MAX_TOP_URLS, items.length));
+}
+
+function sortUrls(urls) {
+    if (urls == null)
+        return [];
+
+	// Create items array
+	var items = Object.keys(urls).map(function(key) {
+		return [key, urls[key]];
+	});
+
+	// Sort the array based on the second element
+	items.sort(function(first, second) {
+		return second[1] - first[1];
+	});
+
+    return items;
+}
+
+function resetDay(d) {
```

Comment:
It would be better to have a "reset" button instead of resetting automatically at midnight.
If you start browsing at 23:00, you lose everything. Would be nice to accummulate things over a longer period of time.

This also requires detecting when the browser is no longer active and stopping the time.